include/dlaf/eigensolver/gen_to_std/mc/gen_to_std_L.h

```diff
@@ -80,21 +84,29 @@ void genToStd_L(Matrix<T, Device::CPU>& mat_a, Matrix<T, Device::CPU>& mat_l) {
         hpx::dataflow(executor_hp, unwrapping(tile::her2k<T, Device::CPU>), Lower, NoTrans, -1.0,
                       mat_a.read(jk), mat_l.read(jk), 1.0, mat_a(jj));
 
-        for (SizeType i = j + 1; i < m; ++i) {
-          const auto ik = LocalTileIndex{i, k};
-          const auto ij = LocalTileIndex{i, j};
-          hpx::dataflow(executor_normal, unwrapping(tile::gemm<T, Device::CPU>), NoTrans, ConjTrans,
-                        -1.0, mat_a.read(ik), mat_l.read(jk), 1.0, mat_a(ij));
-          hpx::dataflow(executor_normal, unwrapping(tile::gemm<T, Device::CPU>), NoTrans, ConjTrans,
-                        -1.0, mat_l.read(ik), mat_a.read(jk), 1.0, mat_a(ij));
+        if (j + 1 < m) {
+          LocalTileIndex istart(j + 1, 0);
+          LocalTileIndex iend(m, 1);
+          for (auto i : dlaf::common::iterate_range2d(istart, iend)) {
+            const auto ik = LocalTileIndex{i.row(), k};
+            const auto ij = LocalTileIndex{i.row(), j};
+            hpx::dataflow(executor_normal, unwrapping(tile::gemm<T, Device::CPU>), NoTrans, ConjTrans,
+                          -1.0, mat_a.read(ik), mat_l.read(jk), 1.0, mat_a(ij));
+            hpx::dataflow(executor_normal, unwrapping(tile::gemm<T, Device::CPU>), NoTrans, ConjTrans,
+                          -1.0, mat_l.read(ik), mat_a.read(jk), 1.0, mat_a(ij));
+          }
```

Comment:
IMHO in this kind of situations, the `iterate_range2d` is not the best choice. To me it reduces drastically the readability of the code.

Reply:
The correct way to use `iterate_range_2d` here is to merge the j (line 80) and i loop (line 90) and select with a if statement when to perform either her2k, gemm or nothing.
Anyway I'm not sure if this improves the readability.